### PR TITLE
Clarify year-of-era issue in test comment

### PR DESCRIPTION
**Pull request by jskeet**
Just to check that the new comment is rather more reasonable than the old one :)

### CODE REVIEW

File: src/NodaTime.Test/Text/LocalDatePatternTest.cs [hunk 1]
```diff
@@ -236,12 +236,15 @@ public void BclLongDatePatternGivesSameResultsInNoda(CultureInfo culture)
...
+            // We got year-of-era and year wrong in 1.x.
             // See https://github.com/nodatime/nodatime/issues/374
-            // It would be a breaking change to make the 2.0 change in 1.x; we could
-            // potentially change the behaviour to allow the given cultures to work
-            // by interpreting YYYY as yyyy if there's no era specifier, but for the moment
-            // these are just expected failures :(
+            // Basically, in 1.x we treat "y" as absolute year and "Y" as year-of-era.
+            // In 2.0 we treat "y" as year-of-era and "u" as absolute year.
+            // In the "known broken cultures" the format contains an era specifier (g)
+            // which we deem to be invalid in 1.x due to the lack of a year-of-era specifier.
```
Comment (malcolmr) [inline src/NodaTime.Test/Text/LocalDatePatternTest.cs]: Slightly confusing: it _does_ have a year-of-era specifier (`y`), just that we think that's an absolute year specifier.

File: src/NodaTime.Test/Text/LocalDatePatternTest.cs [hunk 2]
```diff
@@ -236,12 +236,15 @@ public void BclLongDatePatternGivesSameResultsInNoda(CultureInfo culture)
...
+            // We could potentially fix this for 1.x by only complaining if we're asked to
```
Comment (malcolmr) [inline src/NodaTime.Test/Text/LocalDatePatternTest.cs]: Where "this" is issue #374, I assume (which would then make this workaround unnecessary).

**Comment by malcolmr** [comment]
LGTM in general though.

File: src/NodaTime.Test/Text/LocalDatePatternTest.cs (hunk 1 above)
Comment (jskeet) [inline src/NodaTime.Test/Text/LocalDatePatternTest.cs]: Right. How about "due to what 1.x thinks is a lack of a year-of-era specifier"?

File: src/NodaTime.Test/Text/LocalDatePatternTest.cs (hunk 2 above)
Comment (jskeet) [inline src/NodaTime.Test/Text/LocalDatePatternTest.cs]: Well, sort of. We could potentially fix some symptoms without fixing the "proper" bug (as we have done in 2.0).

File: src/NodaTime.Test/Text/LocalDatePatternTest.cs (hunk 1 above)
Comment (malcolmr) [inline src/NodaTime.Test/Text/LocalDatePatternTest.cs]: sgtm.

**Pull request merged**